pkg/external/machineactuator.go

```diff
@@ -197,19 +209,35 @@ func (c *ExternalClient) waitForJob(jobName string, namespace string, retries in
 								glog.Errorf("failed to delete succeeded job %s: %v", job.Name, err)
 							}
 						}
-						return nil
+						return true, nil
 					}
-					return fmt.Errorf("Job %v failed: %v", job.Name, condition.Message)
+					return false, fmt.Errorf("Job %v failed: %v", job.Name, condition.Message)
 				}
 			}
 		}
 		time.Sleep(5 * time.Second)
 
 		options := metav1.GetOptions{ResourceVersion: job.ObjectMeta.ResourceVersion}
 		job, err = c.kubeclient.BatchV1().Jobs(namespace).Get(job.ObjectMeta.Name, options)
+		if err != nil {
+			return false, err
+		}
+
+		// TODO: when host has OFF state, fence command returns code 2, so I need to catch it
```

Comment:
True. We should distinguish between a status of "off" and "i couldn't connect to the fencing device".
Only applies to status commands though, on/off are smart enough to return success if the node is already in the target state